cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -55,233 +62,240 @@ const F_RETAIN_REDUNDANT_REX: u32 = 1;
 /// indicating a 64-bit operation.
 const F_CLEAR_REX_W: u32 = 2;
 
-/// For specifying the legacy prefixes (or `PfxNone` if no prefix required) to
+/// For specifying the legacy prefixes (or `None` if no prefix required) to
 /// be used at the start an instruction. A select prefix may be required for
 /// various operations, including instructions that operate on GPR, SSE, and Vex
 /// registers.
 enum LegacyPrefix {
-    PfxNone,
-    Pfx66,
-    PfxF2,
-    PfxF3,
+    None,
+    _66,
+    _F2,
+    _F3,
+}
+
+impl LegacyPrefix {
+    #[inline(always)]
+    fn emit(&self, sink: &mut MachBuffer<Inst>) {
+        match self {
+            LegacyPrefix::_66 => sink.put1(0x66),
+            LegacyPrefix::_F2 => sink.put1(0xF2),
+            LegacyPrefix::_F3 => sink.put1(0xF3),
+            LegacyPrefix::None => (),
+        }
+    }
 }
+
 /// This is the core 'emit' function for instructions that reference memory.
 ///
-/// For an instruction that has as operands a register `encG` and a memory
+/// For an instruction that has as operands a register `enc_g` and a memory
 /// address `memE`, create and emit, first the REX prefix, then caller-supplied
-/// opcode byte(s) (`opcodes` and `numOpcodes`), then the MOD/RM byte, then
+/// opcode byte(s) (`opcodes` and `num_opcodes`), then the MOD/RM byte, then
 /// optionally, a SIB byte, and finally optionally an immediate that will be
 /// derived from the `memE` operand.  For most instructions up to and including
 /// SSE4.2, that will be the whole instruction.
 ///
 /// The opcodes are written bigendianly for the convenience of callers.  For
 /// example, if the opcode bytes to be emitted are, in this order, F3 0F 27,
-/// then the caller should pass `opcodes` == 0xF3_0F_27 and `numOpcodes` == 3.
+/// then the caller should pass `opcodes` == 0xF3_0F_27 and `num_opcodes` == 3.
 ///
 /// The register operand is represented here not as a `Reg` but as its hardware
-/// encoding, `encG`.  `flags` can specify special handling for the REX prefix.
+/// encoding, `enc_g`.  `flags` can specify special handling for the REX prefix.
 /// By default, the REX prefix will indicate a 64-bit operation and will be
 /// deleted if it is redundant (0x40).  Note that for a 64-bit operation, the
 /// REX prefix will normally never be redundant, since REX.W must be 1 to
 /// indicate a 64-bit operation.
-fn emit_REX_OPCODES_MODRM_SIB_IMM_encG_memE(
+fn emit_modrm_sib_enc_ge(
```

Comment:
Hmm, this renaming hides the fact that the E operand is memory.  `.._enc_g_mem_e` ?

Reply:
Hmm, my memories of x86 told me that SIB could only be used with a memory operand, so the presence of "sib" in the name implied a memory operand. Besides, I was trying to shorten the name a bit :-) But yeah, I could add it back and it wouldn't be too terrible.